synapse/server.py

```diff
@@ -221,8 +221,6 @@ class HomeServer(metaclass=abc.ABCMeta):
     # instantiated during setup() for future return by get_datastores()
     DATASTORE_CLASS = abc.abstractproperty()
 
-    tls_server_context_factory: Optional[IOpenSSLContextFactory]
-
```

Comment:
I'm not familiar with ABCMeta but I don't think this should have magically an instance attribute?

(What I'm driving at is: why didn't complement fall over on #14128 but did on an earlier version of this PR?)

Reply:
> I'm not familiar with ABCMeta but I don't think this should have magically an instance attribute?

Regardless of `ABCMeta` I would think this would end up as an expected property? But yeah there's no guarantee it actually gets added unless we do it in `__init__`.

(Using a free function to set it is kind of meh...)